theano/compile/ops.py

```diff
@@ -519,6 +519,13 @@ def __str__(self):
         return 'FromFunctionOp{%s}' % self.__fn.__name__
 
     def make_node(self, *inputs):
+
+        if not self.itypes:
+            raise NotImplementedError("itypes not defined")
```

Comment:
NotImplementedError normaly mean that Theano didn't implement that case. As this is a user error, I would think that returning an RuntimeError would be better. Also, I would give a longer error message like:

itypes not defined and make_nodes() not overrieded. In Op subclass, you must override itypes and otypes or overried make_node()